New Pull request for grep cheat sheet

adityabhushan opened the pull request:
- [ ] New Instant Answer
    - [x] Cheat Sheets: **`New Grep Cheat Sheet`**

Instant Answer Page: https://duck.co/ia/view/grep_cheat_sheet

[Maintainer](http://docs.duckduckhack.com/maintaining/guidelines.html): @adityabhushan

daxtheduck commented:
@adityabhushan Hey!
        
Thanks for taking the time to contribute! We really appreciate it.

We work closely with every contributor to make Instant Answers the best they can be, so we appreciate your patience as we look over your code. From here, the process usually goes like this:

1. Pull Request is reviewed by the DuckDuckGo staff and community
2. Staff and community will leave feedback with any necessary updates to the function or design.
3. Once you've made any necessary corrections, then your Instant Answer will be merged and deployed live on DuckDuckGo!

If you have any questions along the way, feel free to ask them here. Our staff and community are also available on Slack to answer any questions you may have. If you'd like to join us there please head to https://quackslack.herokuapp.com/ to get an invite.

>More Info: http://docs.duckduckhack.com/submitting/submitting-overview.html

Thanks!

daxtheduck commented:
Metadata placeholder post
            
*If you haven't already, please edit your first comment to include a link to the IA page at the bottom.*
            
---
            
**How to find your IA page**
- New Instant Answer
  -- follow the steps [here](https://duck.co/ia/new_ia) to create your IA page.
              
- Existing Instant Answer
  -- find your IA page from this [list](https://duck.co/ia)

GuiltyDolphin commented on share/goodie/cheat_sheets/json/grep.json:
```diff
@@ -0,0 +1,174 @@
+    },
+
+    "aliases": [
+        "grep", "grep command", "grep overview"
```
Remove the first alias (it is covered by the cheat sheet's name).

KAUTH commented:
I saw that the checks failed, from a quick look that I gave to the checks' details, you should not have in the "aliases" the same name as the cheat sheet, so no need to put "grep cheat sheet," also check some white spaces and a comma inside of the quotation marks.

tagawa commented on share/goodie/cheat_sheets/json/grep.json:
```diff
@@ -0,0 +1,174 @@
+
+    "metadata": {
+        "sourceName": "",
+        "sourceUrl" : ""
```
Thanks for this cheat sheet @adityabhushan. We should ideally have a source, so please could you use the GNU site: https://www.gnu.org/software/grep/manual/grep.html
(or other reliable website if you prefer)

adityabhushan replied:
Thanks a lot, have modified the alias and  added the suggested source.

On 19 Aug 2016 7:11 am, "Daniel Davis" <notifications@github.com> wrote:

> In share/goodie/cheat_sheets/json/grep.json
> <https://github.com/duckduckgo/zeroclickinfo-goodies/pull/3538#discussion_r75417170>
> :
>
> > @@ -0,0 +1,174 @@
> > +{
> > +    "id": "grep_cheat_sheet",
> > +    "name": "Grep",
> > +    "description": "A quick overview of using the 'grep' command in linux.",
> > +
> > +    "metadata": {
> > +        "sourceName": "",
> > +        "sourceUrl" : ""
>
> Thanks for this cheat sheet @adityabhushan
> <https://github.com/adityabhushan>. We should ideally have a source, so
> please could you use the GNU site: https://www.gnu.org/software/
> grep/manual/grep.html
> (or other reliable website if you prefer)
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/duckduckgo/zeroclickinfo-goodies/pull/3538/files/432e13c85f9a33830e7260b480d1cc8d48bc482c#r75417170>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ACTNl1YlJQpJLCZaAzMAbJyhtCjK61UBks5qhQncgaJpZM4JnMxW>
> .
>

daxtheduck commented:
Automated screenshot from [beta.duckduckgo.com](https://beta.duckduckgo.com/?q=grep cheat sheet)
                ![](https://images.duckduckgo.com/iu/?u=https://ia-screenshots.s3.amazonaws.com/grep_cheat_sheet_index.png?DC0MahmaGS)
                [Mobile screenshot](https://images.duckduckgo.com/iu/?u=https://ia-screenshots.s3.amazonaws.com/grep_cheat_sheet_mobile.png?DC0MahmaGS)

adityabhushan commented:
Thanks a lot :-)

On 19 Aug 2016 1:13 pm, "Dax" <notifications@github.com> wrote:

> Automated screenshot from beta.duckduckgo.com
> <https://beta.duckduckgo.com/?q=grep%20cheat%20sheet>
>
> <https://camo.githubusercontent.com/bfc8030da7b10702ae8f262e3aa377689064a7a9/68747470733a2f2f696d616765732e6475636b6475636b676f2e636f6d2f69752f3f753d68747470733a2f2f69612d73637265656e73686f74732e73332e616d617a6f6e6177732e636f6d2f677265705f63686561745f73686565745f696e6465782e706e673f4443304d61686d614753>
> Mobile screenshot
> <https://images.duckduckgo.com/iu/?u=https://ia-screenshots.s3.amazonaws.com/grep_cheat_sheet_mobile.png?DC0MahmaGS>
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/duckduckgo/zeroclickinfo-goodies/pull/3538#issuecomment-240950586>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ACTNl4Wh1WNZp8PZ7OZBNB4P3REtFTC2ks5qhV6KgaJpZM4JnMxW>
> .
>

tagawa commented:
@adityabhushan Cool, looks like it's working! Let's leave it up on the beta server for a few days to see if there's any feedback from the community: https://beta.duckduckgo.com/?q=grep+cheat+sheet&ia=cheatsheet&iax=1

adityabhushan commented:
Sure Daniel. That seems like a great idea.

On 19 Aug 2016 3:05 pm, "Daniel Davis" <notifications@github.com> wrote:

> @adityabhushan <https://github.com/adityabhushan> Cool, looks like it's
> working! Let's leave it up on the beta server for a few days to see if
> there's any feedback from the community: https://beta.duckduckgo.com/?
> q=grep+cheat+sheet&ia=cheatsheet&iax=1
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/duckduckgo/zeroclickinfo-goodies/pull/3538#issuecomment-240973438>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ACTNl2TR6tnQcbsfT9Xoa9gcou7R3MSQks5qhXjxgaJpZM4JnMxW>
> .
>

jophab commented on share/goodie/cheat_sheets/json/grep.json:
```diff
@@ -0,0 +1,174 @@
+{
+    "id": "grep_cheat_sheet",
+    "name": "Grep",
+    "description": "A quick overview of using the 'grep' command in linux.",
```
Lets remove the period at end of description

jophab commented on share/goodie/cheat_sheets/json/grep.json:
```diff
@@ -0,0 +1,174 @@
+            },
+            {
+                "key": "+",
+                "val": "One of more occurrence of the pattern. "
```
Remove the period here

adityabhushan replied:
Hi, Removed periods from both places.

On 19 Aug 2016 9:48 pm, "JOBIN PHILIP ABRAHAM" <notifications@github.com>
wrote:

> In share/goodie/cheat_sheets/json/grep.json
> <https://github.com/duckduckgo/zeroclickinfo-goodies/pull/3538#discussion_r75510124>
> :
>
> > +                "key": "-E",
> > +                "val": "'Extended Regular Expression' option - Treates pattern as extended regular expression"
> > +            }
> > +        ],
> > +        "RE Wildcards": [
> > +            {
> > +                "key": ".",
> > +                "val": "A single character"
> > +            },
> > +            {
> > +                "key": "*",
> > +                "val": "Zero or more occurances of previous character"
> > +            },
> > +            {
> > +                "key": "+",
> > +                "val": "One of more occurrence of the pattern. "
>
> Remove the period here
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/duckduckgo/zeroclickinfo-goodies/pull/3538/files/a16de752eb86ba3e137ffce3428e533f4d6d6e97#r75510124>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ACTNl-NWwtChJRY6VmNjn9LcOV0vXdpyks5qhddZgaJpZM4JnMxW>
> .
>

daxtheduck commented:
Automated screenshot from [beta.duckduckgo.com](https://beta.duckduckgo.com/?q=grep cheat sheet)
                ![](https://images.duckduckgo.com/iu/?u=https://ia-screenshots.s3.amazonaws.com/grep_cheat_sheet_index.png?2zPmMDngR8)
                [Mobile screenshot](https://images.duckduckgo.com/iu/?u=https://ia-screenshots.s3.amazonaws.com/grep_cheat_sheet_mobile.png?2zPmMDngR8)

daxtheduck commented:
Automated screenshot from [beta.duckduckgo.com](https://beta.duckduckgo.com/?q=grep cheat sheet)
                ![](https://images.duckduckgo.com/iu/?u=https://ia-screenshots.s3.amazonaws.com/grep_cheat_sheet_index.png?11ntRXeXgD)
                [Mobile screenshot](https://images.duckduckgo.com/iu/?u=https://ia-screenshots.s3.amazonaws.com/grep_cheat_sheet_mobile.png?11ntRXeXgD)

tagawa commented on share/goodie/cheat_sheets/json/grep.json:
```diff
@@ -0,0 +1,174 @@
+            },
+            {
+                "key": "-v",
+                "val": "'inverse' option - Displays lines which dont match the expression"
```
Please change `dont` to `don't`

tagawa commented on share/goodie/cheat_sheets/json/grep.json:
```diff
@@ -0,0 +1,174 @@
+            },
+            {
+                "key": "-n",
+                "val": "'number' option - Displays lines numbers along with lineswhich dont match the expression"
```
Please change:
`lines numbers` to `line numbers`
and
`lineswhich` to `lines which`
and
`dont` to `don't`

tagawa commented on share/goodie/cheat_sheets/json/grep.json:
```diff
@@ -0,0 +1,174 @@
+            },
+            {
+                "key": "-E",
+                "val": "'Extended Regular Expression' option - Treates pattern as extended regular expression"
```
Please change `Treates` to `Treats`

tagawa commented on share/goodie/cheat_sheets/json/grep.json:
```diff
@@ -0,0 +1,174 @@
+            },
+            {
+                "key": "*",
+                "val": "Zero or more occurances of previous character"
```
Please change `occurances` to `occurrences`

tagawa commented on share/goodie/cheat_sheets/json/grep.json:
```diff
@@ -0,0 +1,174 @@
+            },
+            {
+                "key": "[xyz]",
+                "val": "Select any one of the character x, y or z"
```
Please change `character` to `characters`

tagawa replied:
Please could you make this change too?

tagawa commented on share/goodie/cheat_sheets/json/grep.json:
```diff
@@ -0,0 +1,174 @@
+            },
+            {
+                "key": "[^a-z]",
+                "val": "Select a single character with does not fall in the ASCII range between 'a' and 'z'"
```
Please change `with` to `which`

tagawa commented on share/goodie/cheat_sheets/json/grep.json:
```diff
@@ -0,0 +1,174 @@
+        "RE Anchors": [
+            {
+                "key": "^",
+                "val": "Begining of a string or line"
```
Please change `Begining` to `Beginning`

tagawa commented on share/goodie/cheat_sheets/json/grep.json:
```diff
@@ -0,0 +1,174 @@
+            },
+            {
+                "key": "\\<",
+                "val": "Begining of word"
```
Please change `Begining` to `Beginning`

tagawa commented on share/goodie/cheat_sheets/json/grep.json:
```diff
@@ -0,0 +1,174 @@
+            },
+            {
+                "key": "grep -i 'first' example.txt",
+                "val": "Returns lines which have the word 'first' or 'First' or 'FIRST' in the file example.txt(returns resules which are case insensitive)"
```
Please change `resules` to `results`, and add a space before the first parenthesis (bracket).

tagawa commented on share/goodie/cheat_sheets/json/grep.json:
```diff
@@ -0,0 +1,174 @@
+            },
+            {
+                "key": "grep 'fi*rst' example.txt",
+                "val": "Returns lines which have the word 'fist', 'first', 'firrst', 'firrrst' etc (with zero or more occunrrences of 'r') in the file example.txt"
```
Please change `occunrrences` to `occurrences`

tagawa commented:
@adityabhushan Thank you for making the changes so quickly. I've just added a few more comments - sorry I didn't spot this before.

adityabhushan commented:
Thanks :-)

Corrected all typos as suggested.

Thanks and Regards,
Aditya Bhushan Dwivedi
+91 72591 83906 | @adi_bhushan <https://twitter.com/adi_bhushan>

On Mon, Aug 22, 2016 at 10:55 AM, Daniel Davis <notifications@github.com>
wrote:

> @adityabhushan <https://github.com/adityabhushan> Thank you for making
> the changes so quickly. I've just added a few more comments - sorry I
> didn't spot this before.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/duckduckgo/zeroclickinfo-goodies/pull/3538#issuecomment-241317588>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ACTNl0Lir9xJpShFXg0BNNhfCpTRhP__ks5qiTKugaJpZM4JnMxW>
> .
>

daxtheduck commented:
Automated screenshot from [beta.duckduckgo.com](https://beta.duckduckgo.com/?q=grep cheat sheet)
                ![](https://images.duckduckgo.com/iu/?u=https://ia-screenshots.s3.amazonaws.com/grep_cheat_sheet_index.png?h0EcIUywl3)
                [Mobile screenshot](https://images.duckduckgo.com/iu/?u=https://ia-screenshots.s3.amazonaws.com/grep_cheat_sheet_mobile.png?h0EcIUywl3)

tagawa commented on share/goodie/cheat_sheets/json/grep.json:
```diff
@@ -0,0 +1,174 @@
+            },
+            {
+                "key": "*",
+                "val": "Zero or more occurences of previous characters"
```
Please change `occurences` to `occurrences` (with 2 "r"s)

daxtheduck commented:
Automated screenshot from [beta.duckduckgo.com](https://beta.duckduckgo.com/?q=grep cheat sheet)
                ![](https://images.duckduckgo.com/iu/?u=https://ia-screenshots.s3.amazonaws.com/grep_cheat_sheet_index.png?w890LKPMcO)
                [Mobile screenshot](https://images.duckduckgo.com/iu/?u=https://ia-screenshots.s3.amazonaws.com/grep_cheat_sheet_mobile.png?w890LKPMcO)

tagawa commented:
Thank you again @adityabhushan! Let's leave it open for a couple more days for any other review comments, then merge after that.

adityabhushan commented:
Sure Daniel.

Thanks for being patient with typos.

Thanks and Regards,
Aditya Bhushan Dwivedi
+91 72591 83906 | @adi_bhushan <https://twitter.com/adi_bhushan>

On Mon, Aug 22, 2016 at 2:15 PM, Daniel Davis <notifications@github.com>
wrote:

> Thank you again @adityabhushan <https://github.com/adityabhushan>! Let's
> leave it open for a couple more days for any other review comments, then
> merge after that.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/duckduckgo/zeroclickinfo-goodies/pull/3538#issuecomment-241349439>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ACTNl06V1lzXmXGovSCATjI4faiTaZraks5qiWG0gaJpZM4JnMxW>
> .
>

tagawa commented:
Heh, no problem! I do it all the time 😃

tagawa commented:
Looks good to me and no further review comments, so let's merge 🎉 
Thank you @adityabhushan for creating this and to @jophab for the feedback (as always!).

And if you'd like to continue contributing, we're currently on a programming mission, aiming to improve Instant Answers for developers. More info on how to help out here: https://forum.duckduckhack.com/t/duckduckhack-programming-mission-overview/53

daxtheduck commented:
@adityabhushan this is now merged! It should be live on DuckDuckGo within one week (or during the next release) and we'll be sure to notify you when it happens.

After your Instant Answer goes live for the World to use, you will be able to see [traffic information](http://docs.duckduckhack.com/maintaining/tools.html#monitor-your-instant-answer-page) about your IA as well as receive feedback directly from users. 
                
Please refer to the [maintenance guide and docs](http://docs.duckduckhack.com/maintaining/guidelines.html) to keep up with potential changes to your Instant Answer.
                
Thank you again for your help!

adityabhushan commented:
Thanks a lot Davis, will keep contributing to the platform. :)

Thanks and Regards,
Aditya Bhushan Dwivedi
+91 72591 83906 | @adi_bhushan <https://twitter.com/adi_bhushan>

On Wed, Aug 24, 2016 at 11:15 AM, Dax <notifications@github.com> wrote:

> @adityabhushan <https://github.com/adityabhushan> this is now merged! It
> should be live on DuckDuckGo within one week (or during the next release)
> and we'll be sure to notify you when it happens.
>
> After your Instant Answer goes live for the World to use, you will be able
> to see traffic information
> <http://docs.duckduckhack.com/maintaining/tools.html#monitor-your-instant-answer-page>
> about your IA as well as receive feedback directly from users.
>
> Please refer to the maintenance guide and docs
> <http://docs.duckduckhack.com/maintaining/guidelines.html> to keep up
> with potential changes to your Instant Answer.
>
> Thank you again for your help!
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/duckduckgo/zeroclickinfo-goodies/pull/3538#issuecomment-241963483>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ACTNlzjg_4Vfx0GXx1_1fi74vWD8ALDMks5qi9p-gaJpZM4JnMxW>
> .
>

tagawa commented:
@adityabhushan I'm pleased to say this is live now - thank you!
https://duckduckgo.com/?q=grep+cheat+sheet&ia=cheatsheet
